api/src/main/java/io/grpc/GlobalInterceptors.java

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright 2022 The gRPC Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.grpc;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.collect.ImmutableList;
+import java.util.List;
+
+/** The collection of global interceptors and global server stream tracers. */
+@Internal
+public final class GlobalInterceptors {
+  private static GlobalInterceptors instance = null;
+  private List<ClientInterceptor> clientInterceptors;
+  private List<ServerInterceptor> serverInterceptors;
+  private List<ServerStreamTracer.Factory> serverStreamTracerFactories;
+  private boolean isGlobalInterceptorsTracersSet;
+  private boolean isGlobalInterceptorsTracersGet;
+
+  @VisibleForTesting
+  GlobalInterceptors() {}
+
+  /** Returns the GlobalInterceptors instance. */
+  private static GlobalInterceptors getInstance() {
```

Comment:
It looks like this and the non-static methods only exists for the purpose of testing. I suggest either making the caller get the singleton so the methods aren't duplicated, or making the fields static and use [StaticTestingClassLoader](https://github.com/grpc/grpc-java/blob/master/context/src/test/java/io/grpc/StaticTestingClassLoader.java). We can discuss in person how the heck that works.

Reply:
Thanks for explaining it in detail. Updated tests to use `StaticTestingClassLoader`.